pkg/controlplane/instance_test.go

```diff
@@ -218,7 +219,7 @@ func TestVersion(t *testing.T) {
 
 	req, _ := http.NewRequest("GET", "/version", nil)
 	resp := httptest.NewRecorder()
-	s.GenericAPIServer.Handler.ServeHTTP(resp, req)
+	s.GenericAPIServer.Handler.ServeHTTP(responsewritertesting.WrapResponseRecorder(resp), req)
```

Comment:
nit: move above:
```
resp := responsewritertesting.WrapResponseRecorder(httptest.NewRecorder())
```

Same in other places

Reply:
Also - can you please move all these changes (including introuction of FakeResponseRecorder) to a separate PR?

Those are needed here, but can be done independently and it would allow us to merge this part early.